cats-backend/src/modules/cats/cats/cats.service.ts

```diff
@@ -17,40 +22,54 @@ export class CatsService {
 
   async getCats(query: { name?: string; mouseName?: string }): Promise<Cat[]> {
     const { name, mouseName } = query;
-    return this.catModel.findAll({
-      attributes: ['id', 'firstName', 'lastName'],
-      where: name
-        ? {
-            [Op.or]: [
-              { firstName: { [Op.iLike]: `%${name}%` } },
-              { lastName: { [Op.iLike]: `%${name}%` } },
-            ],
-          }
-        : undefined,
-      include: [
-        {
-          model: Mouse,
-          attributes: ['id', 'name'],
-          where: mouseName
-            ? {
-                name: {
-                  [Op.iLike]: `%${mouseName}%`,
-                },
-              }
-            : undefined,
-          required: mouseName ? true : false,
+
+    let miceWithCatIds: number[] = [];
+    if (mouseName) {
+      const matchingMice = await this.mouseModel.findAll({
+        attributes: ['catId'],
+        where: {
+          name: { [Op.iLike]: mouseName },
         },
-      ],
+        raw: true,
+      });
+
+      miceWithCatIds = [...new Set(matchingMice.map((mouse) => mouse.catId))];
+      if (!miceWithCatIds.length) return [];
+    }
+
+    const catWhereClause: WhereOptions<Cat> = {};
+
+    if (name) {
+      catWhereClause[Op.or] = [
+        { firstName: { [Op.iLike]: name } },
+        { lastName: { [Op.iLike]: name } },
+      ];
+    }
+
+    if (miceWithCatIds.length > 0) {
+      if (Object.keys(catWhereClause).length > 0) {
+        catWhereClause[Op.and] = [
+          catWhereClause,
+          { id: { [Op.in]: miceWithCatIds } },
```

Comment:
```suggestion
          { id: miceWithCatIds },
```
Optional
Sequlize is smart, it get can a list of ids in id.
But it's the same eventually

Reply:
i tried to change it and it didnt work like this :(